cpp/harvest/core.cc

```diff
@@ -267,62 +271,120 @@ void Download(int argc, char **argv) {
-    if (argc != 5)
+    // if (argc != 5)
+    if (argc != 7)
         Usage();
 
-    bool ignore_unique_id_dups(false);
-    if (std::strcmp(argv[1], "--ignore-unique-id-dups") == 0) {
-        ignore_unique_id_dups = true;
-        --argc, ++argv;
-    }
+    // bool ignore_unique_id_dups(false);
 
     const std::string input_file(argv[2]);
     const std::string output_file(argv[3]);
     const std::string filter_file(argv[4]);
+    const std::string filter_operator(argv[5]);
+    const std::string keyword_file(argv[6]);
+
+    std::string filterKey;
+    std::vector<std::string> filterKeys;
+    std::ifstream fFilter(keyword_file);
+
+    // array constructor
+    while (std::getline(fFilter, filterKey)) {
+        filterKeys.push_back(std::regex_replace(filterKey, std::regex("\\s+$"), std::string("")));
+    }
 
     const auto works(CORE::GetWorksFromFile(input_file));
     CORE::OutputFileStart(output_file);
     CORE::OutputFileStart(filter_file);
     bool first(true);
     unsigned skipped(0), skipped_uni_tue_count(0), skipped_dupe_count(0), skipped_incomplete_count(0), skipped_language_count(0);
-    KeyValueDB unique_id_to_date_map(UNIQUE_ID_TO_DATE_MAP_PATH);
+    // KeyValueDB unique_id_to_date_map(UNIQUE_ID_TO_DATE_MAP_PATH);
     for (auto work : works) {
-        if (work.getPublisher() == "Universität Tübingen") {
-            work.setFilteredReason("Universität Tübingen");
-            CORE::OutputFileAppend(filter_file, work, skipped == 0);
-            ++skipped;
-            ++skipped_uni_tue_count;
-            continue;
-        }
-        if (work.getTitle() == "" or work.getAuthors().empty()) {
-            work.setFilteredReason("Empty title or authors");
-            CORE::OutputFileAppend(filter_file, work, skipped == 0);
-            ++skipped;
-            ++skipped_incomplete_count;
-            continue;
-        }
-
-        static const std::unordered_set<std::string> allowed_languages({ "eng", "ger", "spa", "baq", "cat", "por", "ita", "dut" });
-        if (work.getLanguage().code_.empty() or not allowed_languages.contains(MARC::MapToMARCLanguageCode(work.getLanguage().code_))) {
-            work.setFilteredReason("Language empty or not allowed");
-            CORE::OutputFileAppend(filter_file, work, skipped == 0);
-            ++skipped;
-            ++skipped_language_count;
-            continue;
-        }
+        if (filter_operator == "equal") {
+            if (IsIn(std::to_string(work.getId()), filterKeys)) {
+                std::cout << work.getId() << std::endl;
+                if (work.getPublisher() == "Universität Tübingen") {
+                    work.setFilteredReason("Universität Tübingen");
+                    CORE::OutputFileAppend(filter_file, work, skipped == 0);
+                    ++skipped;
+                    ++skipped_uni_tue_count;
+                    continue;
+                }
+                if (work.getTitle() == "" or work.getAuthors().empty()) {
```

Comment:
There is lots of copy/pasted code here right now. The testing for empty titles/authors, allowed_languages and so on should only be implemented once, regardless of the filter_operator.